File: source/dotnet/Library/AdaptiveCards/AdaptiveFactSet.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
using System.Collections.Generic;
using System.Xml.Serialization;
using Newtonsoft.Json;

namespace AdaptiveCards
{
    /// <summary>
    /// Represents the FactSet element.
    /// </summary>
    [XmlType(TypeName = AdaptiveFactSet.TypeName)]
    public class AdaptiveFactSet : AdaptiveElement
    {
        /// <inheritdoc />
        public const string TypeName = "FactSet";

        /// <inheritdoc />
        [XmlIgnore]
        public override string Type { get; set; } = TypeName;

        /// <summary>
        /// A collection of Facts to this FactSet contains.
        /// </summary>
        [JsonRequired]
        [XmlElement(Type = typeof(AdaptiveFact), ElementName = "Fact")]
        public List<AdaptiveFact> Facts { get; set; } = new List<AdaptiveFact>();
    }
}

File: source/uwp/AdaptiveCardsObjectModel/lib/AdaptiveFactSet.cpp
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
#include "pch.h"

#include "AdaptiveFactSet.h"
#include "AdaptiveFactSet.g.cpp"
#include "AdaptiveFact.h"

namespace winrt::AdaptiveCards::ObjectModel::Uwp::implementation
{
    AdaptiveFactSet::AdaptiveFactSet(const std::shared_ptr<::AdaptiveCards::FactSet>& sharedFactSet)
    {
        Facts = GenerateVectorProjection<implementation::AdaptiveFact>(sharedFactSet->GetFacts());
        InitializeBaseElement(sharedFactSet);
    }

    std::shared_ptr<::AdaptiveCards::BaseCardElement> AdaptiveFactSet::GetSharedModel()
    {
        auto factSet = std::make_shared<::AdaptiveCards::FactSet>();
        CopySharedElementProperties(*factSet);
        factSet->GetFacts() = GenerateSharedVector<implementation::AdaptiveFact, ::AdaptiveCards::Fact>(Facts.get());
        return factSet;
    }
}
